napari/_qt/qt_main_window.py

```diff
@@ -1594,14 +1634,73 @@ def _screenshot(
                     add_flash_animation(self._qt_viewer._welcome_widget)
             finally:
                 # make sure we always go back to the right canvas size
-                if size is not None or scale is not None:
+                if size is not None or scale is not None or fit_to_data_extent:
                     canvas.size = prev_size
+                if fit_to_data_extent:
+                    camera.center = old_center
+                    camera.zoom = old_zoom
```

Comment:
`old_zoom` and `old_center` are defined only in branch. So it is a potential problem with linters. I, personally, prefer to define them and always assign 
```suggestion 
                camera.center = old_center
                camera.zoom = old_zoom
```